### DISCUSSION TITLE
Running a NUE server loop

### DISCUSSION THREAD

**Discussion by Axellarator** [nuejs/nue#39] [Q&A]
Making modifying `create-nue`  a bit easier. I  use Apache as server, but Nginx and other will do the job, too.

1. Create a bash file: `cp-www.sh` 
```
#!/usr/bin/bash
sudo cp -r www/* /var/www/html
```
2. `chmod +x cp-www.sh`
3. Modify your: `package.json` and add the `apache` line
```
  "scripts": {
    "apache": "npm run minify && npm run render && npm run compile && ./cp-www.sh",
```
4.  Put in `layout.nue` this line: 
```
  <meta http-equiv="refresh" content="10"> 
```
5. If you haven't already done so install Inotify: 
```
sudo apt install inotify-tools 
```
6. Use either CLI or bash file:
```
while true; do inotifywait -e modify ./src/ && npm run apache; done
```
Now you will see every change in `src` in your browser with  localhost. Every 10 second Apache does an update. Shorten the refresh to 5 or 1 second, depending on your personal flavor.

**Comment by tipiirai** [comment]
Super cool! Can you think of a hack where both the Apache and the browser reload when there is a file change so that there is no need to wait for n seconds?

**Comment by Axellarator** [comment]
Update: created nue-loop.sh
```
#!/usr/bin/bash

while true; do inotifywait -e modify ./src/ ./www/css/ ./www/img/ && npm run apache; done
```
It covers now the three relevant directories. Script starting point is always the create-nue directory and don't forget chmod +x.
